src/sonic-config-engine/portconfig.py

```diff
@@ -258,6 +259,14 @@ def __init__(self, name, bmode, properties):
 
         if not self._breakout_capabilities:
             raise RuntimeError("Unsupported breakout mode {}!".format(bmode))
+    def _parse_name_to_prefix_and_id(self):
+        match = re.search(swsscommon.FRONT_PANEL_PORT_PREFIX_REGEX + "(\d+)", self._name)
```

Comment:
>FRONT_PANEL_PORT_PREFIX_REGEX [](http://example.com/codeflow?start=37&length=29)

There is an implicit assumption: there must be one and only one group in FRONT_PANEL_PORT_PREFIX_REGEX. This is not flexible and not easy to change in future.

Better to capture the prefix and id into one regex, no concatenating.

Reply:
done